internal/operators/ocs/ocs_operator.go

```diff
@@ -92,6 +94,43 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(context.Context, *common.Cluster, *models.Host) (*models.ClusterHostRequirementsDetails, error) {
-	return &models.ClusterHostRequirementsDetails{}, nil
+func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+	numOfHosts := len(cluster.Hosts)
+
+	if numOfHosts < 3 {
+		return nil, fmt.Errorf("number of hosts is less than 3")
+	} else if numOfHosts == 3 { // Compact Mode
+		if host.Role == models.HostRoleMaster || host.Role == models.HostRoleAutoAssign {
+			return &models.ClusterHostRequirementsDetails{
+				CPUCores:   CPUCompactMode,
+				RAMMib:     MemoryCompactMode,
+				DiskSizeGb: MinDiskSize,
+			}, nil
+		}
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	} else if host.Role == models.HostRoleAutoAssign { //If auto-assign we cannot proceed with minimal or standard mode
+		return nil, fmt.Errorf("unsupported role: %s", host.Role)
+	} else {
+		// Minimum mode is the minimum version of the standard mode. If the resources will be more, standard mode will be deployed automatically.
+		// In minimum and standard mode we don't care about masters so returning zero
+		if host.Role == models.HostRoleMaster {
+			return &models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0}, nil
+		}
+
+		var inventory *models.Inventory
```

Comment:
you can use this function [here](https://github.com/openshift/assisted-service/blob/master/internal/host/hostutil/host_utils.go#L109) instead, it does the trick :wink:
Also, not a good idea to ignore an error coming from a function :smile:

Reply:
+1